master/internal/core_experiment.go

```diff
@@ -291,6 +291,15 @@ func (m *Master) parseCreateExperiment(ctx context.Context, req *apiv1.CreateExp
 	if err != nil && errors.Cause(err) != sql.ErrNoRows {
 		return nil, nil, config, nil, nil, err
 	}
+
+	// ensure project and workspace name is set, even if specified via req
+	if p.Id > 1 {
```

Comment:
This ensures that experiments created with `--project_id <ID>` have the workspace name added to the task spec